### PR TITLE
Microgen [WIP]

### PR DESCRIPTION
**Pull request by auag92**
Still working on the pull request...
Will update when it is ready for review.

### CODE REVIEW

**Comment by wd15** [comment]
@auag92, PR needs to merge into to develop, not master

**Comment by auag92** [comment]
> @auag92, PR needs to merge into to develop, not master

Thanks Daniel. I have amended the pull request.

**Comment by beyucel** [comment]
Hey Apaar,  you also need to push only  `pymks/fmks/data/multiphase.py`  and ` notebooks/intro.ipynb`. We are not pushing other files (domain.vtk and daskworker space etc.)

**Comment by beyucel** [comment]
@wd15 do you have any idea why travis ci tests were not performed after Apaar change PR to develop?

**Comment by wd15** [comment]
> @wd15 do you have any idea why travis ci tests were not performed after Apaar change PR to develop?

Changing the destination branch doesn't change the contents of the PR so there is no reason to rerun the tests.

File: domain.vtk [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# vtk DataFile Version 2.0
```
Comment (wd15) [inline domain.vtk:1]: This shouldn't be in the PR

**Comment by auag92** [comment]
Okay... Daniel...
I will do that and get back.

Its still a draft PR. I will assign reviewers when its ready.

regards,
Apaar Shanker

*---------------------------------------------------------------------*

*College of ComputingGeorgia Institute of Technology*

**Comment by beyucel** [comment]
> > @wd15 do you have any idea why travis ci tests were not performed after Apaar change PR to develop?
> 
> Changing the destination branch doesn't change the contents of the PR so there is no reason to rerun the tests.

It makes sense, but just for curiosity, the previous destination did not have .travis.yaml file and the new one has. Shouldn't it trigger the Travis tests?

**Comment by beyucel** [comment]
@auag92 @wd15   Also, this is for microstructure generation. To have all the functionalities of `make_elastic_stress_random`, we need to call sfepy as well. Are we going to do that on notebooks ( as we did for plotting) or do we need to write another FE_simulation.py file?

**Comment by wd15** [comment]
> It makes sense, but just for curiosity, the previous destination did not have .travis.yaml file and the new one has. Shouldn't it trigger the Travis tests?

"master" does have a .travis.yml, but the CI uses the .travis.yml in the branch that is being merged, not that which is being merged onto.

**Comment by wd15** [comment]
> @auag92 @wd15 Also, this is for microstructure generation. To have all the functionalities of `make_elastic_stress_random`, we need to call sfepy as well. Are we going to do that on notebooks ( as we did for plotting) or do we need to write another FE_simulation.py file?

Yes. That's how I did it for the delta microstructure FE simulations. It makes much more sense.

**Pull request closed without merging**